composition/test/test_api_srv_composition_client_first.py.in

```diff
@@ -12,69 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import unittest
+
 from launch import LaunchDescription
-from launch import LaunchService
 from launch.actions import ExecuteProcess
-from launch_testing import LaunchTestService
-from launch_testing.output import create_output_lines_filter
-from launch_testing.output import create_output_test_from_file
-
+from launch.actions import OpaqueFunction
 
-def test_api_srv_composition_client_first():
-    # This is a regression test to ensure that the client starting first and waiting for a service
-    # does not block the executor from processing requests to load other components.
-    name = 'test_api_composition'
-    executable = '@API_COMPOSITION_EXECUTABLE@'
-    additional_processes = [
-        {
-            'cmd': [
-                '@API_COMPOSITION_CLI_EXECUTABLE@',
-                'composition', 'composition::Client'],
-            'name': 'load_client_component',
-        },
-        {
-            'cmd': [
-                '@API_COMPOSITION_CLI_EXECUTABLE@',
-                'composition', 'composition::Server', '--delay', '5000'],
-            'name': 'load_server_component',
-        },
-    ]
-    output_file = '@EXPECTED_OUTPUT_SRV@'
-    launch(name, [executable], output_file, additional_processes=additional_processes)
+import launch_testing
+import launch_testing.asserts
+import launch_testing_ros
 
 
-def launch(name, cmd, output_file, additional_processes=None):
-    launch_test = LaunchTestService()
+def generate_test_description(ready_fn):
     launch_description = LaunchDescription()
-
-    action = launch_test.add_fixture_action(
-        launch_description, ExecuteProcess(
-            cmd=cmd,
-            name=name,
-            output='screen'
-        )
+    process_under_test = ExecuteProcess(
+        cmd=['@API_COMPOSITION_EXECUTABLE@'],
+        name='test_api_composition',
+        output='screen'
     )
-    rmw_implementation = '@rmw_implementation@'
-    launch_test.add_output_test(
-        launch_description, action,
-        output_test=create_output_test_from_file(output_file),
-        output_filter=create_output_lines_filter(
-            filtered_rmw_implementation=rmw_implementation
+    launch_description.add_action(process_under_test)
+    launch_description.add_action(
+        ExecuteProcess(
+            cmd=[
+                '@API_COMPOSITION_CLI_EXECUTABLE@',
+                'composition', 'composition::Client'
+            ],
+            name='load_client_component'
         )
     )
-    for additional_process in (additional_processes or []):
-        launch_test.add_fixture_action(
-            launch_description, ExecuteProcess(**additional_process)
+    launch_description.add_action(
+        ExecuteProcess(
+            cmd=[
+                '@API_COMPOSITION_CLI_EXECUTABLE@',
+                'composition', 'composition::Server',
+                '--delay', '5000'
```

Comment:
If the API_COMPOSITION_CLI_EXECUTABLE client has some stdout, you can delay the API_COMPOSITION_CLI_EXECUTABLE server  start using a [stdout ready listener](https://github.com/ros2/launch/blob/master/apex_rostest/apex_launchtest/apex_launchtest/event_handlers/stdout_ready_listener.py) instead of a blind 5s wait - it may speed up your test

Reply:
Sounds reasonable, but I'd rather not touch working tests just yet.